numpy/lib/function_base.py

```diff
@@ -2530,13 +2542,21 @@ def corrcoef(x, y=None, rowvar=1, bias=np._NoValue, ddof=np._NoValue):
     c = cov(x, y, rowvar)
     try:
         d = diag(c)
-    except ValueError:  # scalar covariance
+    except ValueError:
+        # scalar covariance
         # nan if incorrect value (nan, inf, 0), 1 otherwise
         return c / c
-    d = sqrt(d)
-    # calculate "c / multiply.outer(d, d)" row-wise ... for memory and speed
-    for i in range(0, d.size):
-        c[i,:] /= (d * d[i])
+    stddev = sqrt(d.real)
+    c /= stddev[:, None]
+    c /= stddev[None, :]
```

Comment:
It's about the same. Using multiplication makes it 2% faster for the test problem in the original fix. I think most of the time is always going to be in `cov`. However, I prefer this version for its clarity, OTOH, you get two zero division warnings when one of the diagonals is zero (which can happen).